markriverac/academy-go-q42021
# Adding concurrency endpoint

## Adding concurrency

### How did you implement it

1. Added a channel where pokemon items are being saved.
2. Create a worker by dividing the `items` param by the `items_per_worker` param.
3. Each worker validates the type of the pokemon (if it is `even` or `odd`) and adds the result to a result channel.
4. When the workers end their work the results channel is sent to the response.

### How to test it

1. Run`go run main/main.go`
2. In your browser go to the route `http://localhost:10000/concurrency?type=odd&items=50&items_per_worker=1` and set the values that you'd like for the query params.
3. You should see a list with the results in a `json` format.
